### DISCUSSION TITLE
g++ error: no such file or directory (To build with MONAI Cpp/CUDA extensions)

### DISCUSSION THREAD

**Discussion by hired-ld** [Project-MONAI/MONAI#3200] [Q&A]
To build with MONAI Cpp/CUDA extensions, I ran codes:
git clone https://github.com/Project-MONAI/MONAI.git
cd MONAI/
BUILD_MONAI=1 python setup.py develop

but, some errors happened. How to solve this question?

g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/ext.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/resample/pushpull_cpu.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/filtering/bilateral/bilateral.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/filtering/bilateral/bilateralfilter_cpu.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/filtering/bilateral/bilateralfilter_cpu_phl.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/filtering/permutohedral/permutohedral_cpu.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/filtering/permutohedral/permutohedral.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/lltm/lltm_cpu.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/resample/pushpull_cuda.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/filtering/bilateral/bilateralfilter_cuda.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/filtering/bilateral/bilateralfilter_cuda_phl.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/filtering/permutohedral/permutohedral_cuda.o: 没有那个文件或目录
g++: error: /home/.../MONAI/build/temp.linux-x86_64-3.6/home/.../MONAI/monai/csrc/lltm/lltm_cuda.o: 没有那个文件或目录

**Comment by Nic-Ma** [comment]
Hi @wyli ,

I think you know more details about the extensions, could you please help share some comments here?

Thanks in advance.

**Comment by wyli** [comment]
(addressed in https://github.com/Project-MONAI/MONAI/issues/3201)
